modules/json/src/smithy4s/http/json/codecs.scala

```diff
@@ -37,7 +37,9 @@ object codecs {
       Discriminated,
       Untagged,
       InputOutput,
-      DiscriminatedUnionMember
+      DiscriminatedUnionMember,
+      // TODO: add tests for `codecs` understanding int enums. Maybe pizza spec
+      IntEnum
```

Comment:
**issue:** Must've slipped through the cracks, but I think because this was missing the HTTP codecs would never encode/decode an enum as an int. This will affect dynamic _and_ codegen.

@daddykotex would you be so kind and implement a test for this? Might be PizzaSpec+PizzaClientSpec worthy, but probably just checking it in a http4s spec would be good enough

Reply:
will do